doc/crypt.tex

```diff
@@ -5507,6 +5507,22 @@ \subsection{Signature Generation}
 This function creates the same ECDSA signature as \textit{ecc\_sign\_hash} only the output format is different.
 The format follows \url{https://tools.ietf.org/html/rfc7518#section-3.4}, sometimes it is also called plain signature.
 
+\index{ecc\_sign\_hash_ex()}
+\begin{verbatim}
+int ecc_sign_hash_ex(const unsigned char *in,
+                           unsigned long  inlen,
+                           unsigned char *out,
+                           unsigned long *outlen,
+                              prng_state *prng,
+                                     int  wprng,
+                                     int  sigformat,
+                                     int *recid,
+                                 ecc_key *key);
+\end{verbatim}
+
+This function is an extended version of the ECDSA signature in \textit{ecc\_sign\_hash}, but with a choice of output formats
+and an optional output of the recovery ID for use with \textit{ecc\_recover\_key}.
+
```

Comment:
a table with supported `sigformat` values would be handy here:

| sigformat | description |
|-------|-------------|
|  LTC_ECCSIG_ANSIX962 | ASN.1 encoded, ANSI X9.62 |
|  LTC_ECCSIG_RFC7518  | raw R, S values as defined in RFC7518 |
| LTC_ECCSIG_BLOCKCHAIN | raw V, R, S values |

Reply:
I've not done any LaTeX before this, so I'm not familiar with how it all works... something like this?

```
\begin{figure}[hpbt]
\index{Signature Formats}
\begin{small}
\begin{center}
\begin{tabular}{|l|l|}
\hline \textbf{sigformat} & \textbf{description} \\
\hline LTC_ECCSIG_ANSIX962 & ASN.1 encoded, ANSI X9.62 \\
\hline LTC_ECCSIG_RFC7518 & raw R, S values as defined in RFC7518 \\
\hline LTC_ECCSIG_ETH & raw V, R, S values as defined in Yellow Paper\\
\hline
\end{tabular}
\end{center}
\end{small}
\caption{Signature Formats}
\label{fig:sigformat}
\end{figure}
```

Reply:
you have to escape `_` as `\_`

```
\begin{figure}[hpbt]
\index{Signature Formats}
\begin{small}
\begin{center}
\begin{tabular}{|l|l|}
\hline \textbf{sigformat} & \textbf{description} \\
\hline LTC\_ECCSIG\_ANSIX962 & ASN.1 encoded, ANSI X9.62 \\
\hline LTC\_ECCSIG\_RFC7518 & raw R, S values as defined in RFC7518 \\
\hline LTC\_ECCSIG\_ETH & raw V, R, S values as defined in Yellow Paper\\
\hline
\end{tabular}
\end{center}
\end{small}
\caption{Signature Formats}
\label{fig:sigformat}
\end{figure}
```

Reply:
Cheers!
I'd got a mistake before with an unescaped _ in `CRYPT_OK`, so I've fixed that, too. Managed to get a build of the doc working - LaTeX isn't fun on a Mac - and it all looks fine now.